bazel/google_cloud_cpp_deps.bzl

```diff
@@ -116,11 +116,11 @@ def google_cloud_cpp_deps(name = None):
         http_archive,
         name = "com_google_protobuf",
         urls = [
-            "https://storage.googleapis.com/cloud-cpp-community-archive/com_google_protobuf/v21.12.tar.gz",
-            "https://github.com/protocolbuffers/protobuf/archive/v21.12.tar.gz",
+            "https://storage.googleapis.com/cloud-cpp-community-archive/com_google_protobuf/v23.1.tar.gz",
+            "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v23.1.tar.gz",
```

Comment:
Should we drop the "refs/tags/" (et seq)?

Reply:
Done.  FWIW, I think it makes no difference.  Both redirect to the same URL.